sensor_msgs_py/sensor_msgs_py/point_cloud2.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc. (2020)
+# All rights reserved.
+#
+# Software License Agreement (BSD License 2.0)
```

Comment:
This is totally not your fault, but we should fix up this license header.  This version of the header is a kind of mixed-up BSD 3-clause license.  The "official" license header is here: https://github.com/ament/ament_lint/blob/master/ament_copyright/ament_copyright/template/bsd_3clause_header.txt

(the same goes for all of the headers below)

The other problem we now have in this repository is that there is a top-level LICENSE file that specifies Apache 2.0, but that is no longer true for the entire repository.  We could do one of two things here:

1.  Rename the top-level LICENSE file to LICENSE.apache2 , and then add in LICENSE.bsd3 alongside it.
1.  Copy the current Apache 2 LICENSE file to each of the existing packages, then add in a new LICENSE file to this package with the BSD license in it.

I'm kind of leaning more towards the second option, but I'd like to hear any additional opinions from @ros2/team

Reply:
@clalancette wow that license header thing is frustrating. i didnt know there was two different bsd 3 clause licenses in ament_copyright. why is there two different of the same license? and why is one "slightly" more right?

Reply:
> @clalancette wow that license header thing is frustrating. i didnt know there was two different bsd 3 clause licenses in ament_copyright. why is there two different of the same license? and why is one "slightly" more right?

When I originally added it, I made a mistake and put slightly wrong text in there.  We've been slowly deprecating and removing that slightly incorrect version, but we haven't totally gotten rid of it yet.  So it is still there for legacy reasons; one of these days we'll remove it.  In any case, we shouldn't add new code that depends on it.

Reply:
@clalancette why not just change this package to Apache 2.0 as well to match the rest of the packages in this repo? seems kind of arbitrary to change only this package to BSD 2.0.

Reply:
> @clalancette why not just change this package to Apache 2.0 as well to match the rest of the packages in this repo? seems kind of arbitrary to change only this package to BSD 2.0.

Since this is a port of existing functionality, we can't unilaterally do that unless we have agreement from all of the previous authors.

Reply:
@clalancette it looks like from [the original file](https://github.com/ros/common_msgs/blame/noetic-devel/sensor_msgs/src/sensor_msgs/point_cloud2.py) that @tfoote  or @dirk-thomas  could sign-off and say if we can switch to Apache 2.0 like the rest of this repo.

could either @tfoote or @dirk-thomas  sign off and agree that this package could be Apache 2.0 now to match the parent repo?

Reply:
@sebastiangrans ive made another [PR to your branch](https://github.com/SebastianGrans/common_interfaces/pull/2) that switches this repo to Apache 2.0 to match the parent repo.

Reply:
@clalancette should the copyright name then still be kept as the original `Willow Garage, Inc.`?

Reply:
Oh yeah, good point.  It should be kept as the original then.

Reply:
@clalancette @sebastiangrans PR [open here](https://github.com/SebastianGrans/common_interfaces/pull/4) with the discussed changes. i think i got them all...but definitely still check it

Reply:
That looks reasonable to me.  As long as the linters pass, I'm happy with it.  We just need @SebastianGrans to merge that into this PR, then we can move forward here.

Reply:
Merged! :)